### ISSUE TITLE
Just-in-Time Provisioning failing  (SDK V4_Beta)

### ISSUE THREAD

**Issue by shaahad** [aws/aws-iot-device-sdk-embedded-C#312]
Hi, 

The new SDK is failing to produce a thing in the server automatically, during the just-in-time provisioning. 

Using the exact same generated certs on my device the two SDK's behave differently

SDK master: 
-----------------
MQTT connection establish and it creates a thing in the server automatically and my device is connected.

SDK V4_Beta: 
-------------------
MQTT connection fails with a timeout error and NO "Thing" gets created in the server.The log messages I can see that the TLS handshake is successful and it fails after the TLS handshake. Would you be able to highlight if we need to configure something to enable the SDK to create the thing automatically.

NOTE: No issue with Manual creation of Thing and generating the certs.
 
please find 
[iotSDK_v4Beta_MQTTConnectionFailure.log](https://github.com/aws/aws-iot-device-sdk-embedded-C/files/2962233/iotSDK_v4Beta_MQTTConnectionFailure.log)

**Comment by gordonwang0** [comment]
Hi @shaahad,

Thanks for bringing this up.  I haven't tested JIT provisioning yet on v4.  I'll get back to you in a few days about this.

**Comment by gordonwang0** [comment]
Hi @shaahad,

I believe I've found the cause of this issue.

https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/11f7d6373b278809fb72822e665e5ef589b78738/platform/source/posix/network/iot_network_openssl.c#L454-L456

JITP should work if this line is changed to:
```c
if( SSL_CTX_use_certificate_chain_file( pSSLContext, pClientCertPath ) != 1 )
```

Please give this a try if you need an immediate fix.  I'll make this fix after we're done with integrating some of the latest updates.

(Note that OpenSSL requires your certificate chain file to be in the order: client cert, root CA cert.)

**Comment by shaahad** [comment]
Hi @gordonwang0, 

Thank you for your response. 

I have implemented the change into my local copy and I can see the "Thing" gets created in the server while running "IotMqtt_Connect". However, "IotMqtt_Connect" returns timeout error.

I have attached the log below.
[iotSDK_v4Beta_MQTTConnectionFailure.log](https://github.com/aws/aws-iot-device-sdk-embedded-C/files/2982651/iotSDK_v4Beta_MQTTConnectionFailure.log)

I also noticed when I run my application again I get a connection established successful. I believe it connects because the "Thing" is already created in the server as shown in the log.
[iotSDK_v4Beta_SuccessfulConnection.log](https://github.com/aws/aws-iot-device-sdk-embedded-C/files/2982663/iotSDK_v4Beta_SuccessfulConnection.log)
